sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestForm.java

```diff
@@ -380,7 +396,7 @@ protected void addFields() {
 			setCqValueVisibility(cqValueField, (PathogenTestType) testTypeField.getValue(), testResult);
 		});
 
-		if (SamplePurpose.INTERNAL.equals(getSamplePurpose())) { // this only works for already saved samples
+		if (environmentSample == null && SamplePurpose.INTERNAL.equals(getSamplePurpose())) { // this only works for already saved samples
```

Comment:
better to update `getSamplePurpose` instead of extending the condition here

Reply:
removed and updated getSamplePurpose